client/src/components/Leaflet.js

```diff
@@ -242,6 +271,83 @@ const Leaflet = ({
     widthPropUnchanged
   ])
 
+  const { data } = API.useApiQuery(allLocationsQuery)
```

Comment:
we could retrieve this actually only once, like when the module is loaded, and share the content across all map instances